splunk_otel/metrics.py

```diff
@@ -0,0 +1,51 @@
+# Copyright Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+
+from opentelemetry.exporter.otlp.proto.grpc.metric_exporter import OTLPMetricExporter
+from opentelemetry.instrumentation.system_metrics import SystemMetricsInstrumentor
+from opentelemetry.metrics import set_meter_provider
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader
+
+from splunk_otel.util import _is_truthy
+
+logger = logging.getLogger(__file__)
+logger.setLevel(logging.INFO)
+
+
+def start_metrics() -> MeterProvider:
+    enabled = os.environ.get("OTEL_METRICS_ENABLED", True)
+    if not _is_truthy(enabled):
+        logger.info("metering has been disabled with OTEL_METRICS_ENABLED=%s", enabled)
+        return None
+
+    try:
+        meter_provider = _configure_metrics()
+        system_metrics = SystemMetricsInstrumentor()
+        system_metrics.instrument(meter_provider=meter_provider)
+        logger.debug("Instrumented runtime metrics")
```

Comment:
Since the logger's level is set to `INFO` earlier, this line does nothing.

Reply:
Kept it the same but added the env variable for the log level so this line would show up if the log level is set via the env.